node-jose/index.d.ts

```diff
@@ -0,0 +1,1380 @@
+     */
+    type encOpts = {
+        zip?: boolean | string,
+        format?: string,
+        compact?: boolean,
+        contentAlg?: string,
+        protect?: string | string[],
+        fields?: any,
+        aad?: string,
+        cek?: string,
+        iv?: string
+    };
+
+    /**
+     * Factory function that reates a new Decrypter for the given Key or KeyStore.
+     *
+     * @param {(JWK.KeyStore|JWK.Key)} ks The Key or KeyStore to use for decryption.
+     * @param {decryptCritHeaderOptions} [opts] The options for this decryptor.
+     * @returns {Decrypter} The new Decryptor.
+     */
+    function createDecrypt(ks: JWK.KeyStore | JWK.Key, opts?: decryptCritHeaderOptions): Decrypter;
+
+    /**
+     * Factory function to create `Encrypter`s.  See doc for opts and return for more info.
+     *
+     * @param {enc_opts} opts The options used to create the Encrypter.
+     * @param {(JWK.Key|JWK.Key[]|{key: JWK.Key}|{key: JWK.Key}[])} rcpts The recipient list for the JWT
+     * @returns {Encrypter} An object that can perform actions turning JWTs into JWEs.
+     */
+    function createEncrypt(opts: encOpts, rcpts: JWK.Key | JWK.Key[] | { key: JWK.Key } | { key: JWK.Key }[]): Encrypter;
+    function createEncrypt(rcpts: JWK.Key | JWK.Key[] | { key: JWK.Key } | { key: JWK.Key }[]): Encrypter;
+
+    /**
+     * @class Decrypter
+     * @classdesc
+     * Processor of encrypted data.
+     *
+     * @description
+     * **NOTE:** This class cannot be instantiated directly. Instead
+     * call {@link createDecrypt}.
+     */
+    interface Decrypter {
+
+        /**
+         * Decrypts the given input.
+         *
+         * {opts}, if provided, is used to customize this specific decrypt operation.
+         * This argument has the same semantics as {JWE.createDecrypt}, and takes
+         * precedence over those options.
+         *
+         * @param {object|string} input The encrypted content
+         * @param {decryptCritHeaderOptions} [opts] The options for this decryption operation.
+         * @returns {Promise<decryptReturn>} A promise for the decyprted plaintext
+         * @memberOf Decrypter
+         */
+        decrypt(input: object | string, opts?: decryptCritHeaderOptions): Promise<decryptReturn>;
+    }
+
+    /**
+     * @class Encrypter
+     * @classdesc
+     * Generator of encrypted data.
+     *
+     * @description
+     * **NOTE:** This class cannot be instantiated directly. Instead call
+     * {@link createEncrypt}.
+     */
+    interface Encrypter {
+        /**
+         * Indicates the compression algorithm applied to the plaintext
+         * before it is encrypted.  The possible values are:
+         *
+         * + **`"DEF"`**: Compress the plaintext using the DEFLATE algorithm.
+         * + **`""`**: Do not compress the plaintext.
+         *
+         * @type {string}
+         * @memberOf Encrypter
+         * @readonly
```

Comment:
Also the type should be `"DEF" | ""`, which is more specific than just `string`.